server/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -149,6 +150,12 @@ public void testApplyDataFromRequest() throws Exception {
         assertThat(getMappingsFromResponse(), Matchers.hasKey("mapping1"));
     }
 
+    public void testInvalidAliasName() throws Exception {
+        final String[] invalidAliasNames = new String[] { "-alias1", "+alias2", "_alias3", "a#lias", "al:ias", ".", ".." };
+        setupRequestAlias(new Alias(randomFrom(invalidAliasNames)));
+        expectThrows(InvalidAliasNameException.class, this::executeTask);
+    }
```

Comment:
This is testing that aliases with names with leading '-' cannot indeed be created so that `-*` can be safely used as an exclude wildcard (as it presently is).